source/textInfos/__init__.py

```diff
@@ -501,32 +512,54 @@ def getTextInChunks(self, unit):
 			yield chunkInfo.text
 			unitInfo.collapse(end=True)
 
-	def getControlFieldSpeech(self, attrs, ancestorAttrs, fieldType, formatConfig=None, extraDetail=False, reason=None):
+	def getControlFieldSpeech(
+			self,
+			attrs: ControlField,
+			ancestorAttrs: List[Field],
+			fieldType: str,
+			formatConfig: Optional[Dict[str, bool]] = None,
+			extraDetail: bool = False,
+			reason: Optional[str] = None
+	) -> SpeechSequence:
 		# Import late to avoid circular import.
 		import speech
-		return speech.getControlFieldSpeech(attrs, ancestorAttrs, fieldType, formatConfig, extraDetail, reason)
+		sequence = speech.getControlFieldSpeech(
+			attrs, ancestorAttrs, fieldType, formatConfig, extraDetail, reason
+		)
+		_logBadSequenceTypes(sequence)
+		return sequence
 
 	def getControlFieldBraille(self, field, ancestors, reportStart, formatConfig):
 		# Import late to avoid circular import.
 		import braille
 		return braille.getControlFieldBraille(self, field, ancestors, reportStart, formatConfig)
 
-	def getFormatFieldSpeech(self, attrs, attrsCache=None, formatConfig=None, reason=None, unit=None, extraDetail=False , initialFormat=False, separator=None):
+	def getFormatFieldSpeech(
+			self,
+			attrs: Field,
+			attrsCache: Optional[Field] = None,
+			formatConfig: Optional[Dict[str, bool]] = None,
+			reason: Optional[str] = None,
+			unit: Optional[str] = None,
+			extraDetail: bool = False,
+			initialFormat: bool = False,
+	) -> SpeechSequence:
 		"""Get the spoken representation for given format information.
 		The base implementation just calls L{speech.getFormatFieldSpeech}.
 		This can be extended in order to support implementation specific attributes.
 		If extended, the superclass should be called first.
-		@param separator: The text used to separate chunks of format information;
-			defaults to L{speech.CHUNK_SEPARATOR}.
-		@type separator: str
 		"""
 		# Import late to avoid circular import.
 		import speech
-		if separator is None:
-			# #6749: The default for this argument is actually speech.CHUNK_SEPARATOR,
-			# but that can't be specified as a default argument because of circular import issues.
-			separator = speech.CHUNK_SEPARATOR
-		return speech.getFormatFieldSpeech(attrs, attrsCache=attrsCache, formatConfig=formatConfig, reason=reason, unit=unit, extraDetail=extraDetail , initialFormat=initialFormat, separator=separator)
+		return speech.getFormatFieldSpeech(

```

Comment:
May be you want to log bad sequences here as well, as in getControlFieldSpeech?

Reply:
speech.getFormatFieldSpeech logs it. If this method augmented the sequence it should probably check. For now I think this is fine.